### DISCUSSION TITLE
Provide an option to not render .rst files

### DISCUSSION THREAD

**Discussion by WarrenWeckesser** [community/community#7999] [Other Feature Feedback, Questions, & Ideas]
When I click on a file with the extension `.md`, I get a nicely rendered view of the file. There are buttons at the top that allow me to either "Display the source blob" or "Display the rendered blob". By viewing the source blob, I can inspect the markdown source code of the documentation. This is an important feature for any project that uses markdown as the source code of their documentation.

Could you please add the same buttons for files with the `.rst` extension?  Currently, such files are *always* rendered, with no option to view the source code as a regular text files. Many projects (including two for which I am a developer, NumPy and SciPy) use `.rst` files as the source code of their documentation. It is important to be able to view that source code without it being rendered.

**Comment by NickleDave** [comment]
:+1: 

Having that button for .rst would be great

I actually just figured out yesterday that you can get .rst to render as "source blob" by adding "?plain=1" to the url as described in this 2014 blog post
https://github.blog/changelog/2021-06-30-parameter-to-disable-markdown-rendering/
but obviously just clicking on a UI element would be much friendlier!

And there are a lot of Python docs in .rst out there
:pray: would be much appreciated

**Reply by mankoff** [reply]
I can't select line numbers and open an issue against them, the way I can with the GitHub markdown "view source blob" feature.

**Reply by andreasabel** [reply]
> I can't select line numbers and open an issue against them, the way I can with the GitHub markdown "view source blob" feature.

Works for me with the workaround `?plain=1`, but it is cumbersome.

**Reply by mankoff** [reply]
Yup - you're correct. Not sure what I was doing wrong.

**Comment by WarrenWeckesser** [comment]
I just ran into this again, and was reminded how much of a nuisance this is.

@NickleDave's suggestion is helpful, but it sure would be nice to get this fixed.  A consistent user interface is important--*any* file type that is automatically rendered should have the same buttons that are available for markdown.   And the ability to treat `.rst` files as source code instead of being rendered is important for any project that uses `.rst` files as the source of their documentation.

**Comment by mankoff** [comment]
Also Org files. Any rendered markdown should have a "view source blob" option.
